File: app/src/main/java/eu/indiewalkabout/fridgemanager/ui/FoodsViewModelFactory.java
package eu.indiewalkabout.fridgemanager.ui;

import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;
import androidx.annotation.NonNull;

public class FoodsViewModelFactory extends ViewModelProvider.NewInstanceFactory {

    private final String foodlistType;

    public FoodsViewModelFactory(String foodlistType) {
        this.foodlistType = foodlistType;
    }

    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        return (T) new FoodsViewModel(foodlistType);
    }
}

File: app/src/main/java/eu/indiewalkabout/fridgemanager/ui/FoodsViewModelFactory.kt
package eu.indiewalkabout.fridgemanager.ui

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider

class FoodsViewModelFactory(private val foodlistType: String) : ViewModelProvider.NewInstanceFactory() {
    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
        return FoodsViewModel(foodlistType) as T
    }

}